### PR TITLE
Feature: Separation of ADMIN and SUPERADMIN functionalities

### LINKED ISSUE #650: Isolate Admins from Viewing any Super Admin pages
**Is your feature request related to a problem? Please describe.**
Admins of several organizations may control the personnel and operations of their own organizations using talawa-admin. Presently, when you attempt to connect into talawa-admin, you are logged in as super-admin and are able to access information for other organizations in addition to your own.
You can refer to the image below.

**Describe the solution you'd like**
1. The login screen should stay the same, but when a user enters their credentials, it should automatically determine if they are an admin or a super-admin. 
1. Although the super admin might travel down through all the organizations, the admin should only be allowed access and view their specific organization.
1. No screens accessible only to Super Admins must be visible to Admins.
1. If an Admin gets sent a link to a Super Admin page while they are logged in as an Admin, then there should be no access provided.

This is an extension of this issue:
- https://github.com/PalisadoesFoundation/talawa-admin/issues/493

**Current  Behaviour**

https://user-images.githubusercontent.com/16875803/224429182-092ce9ca-c0ff-4ce3-9202-a8e1c4b01ed6.mp4

**Describe alternatives you've considered**
The contributor can come up with her/ his own ideas.

**Approach to be followed**
- We want to minimize changes to the existing design. Keep the current design.
- Code coverage for all files editor must be 100%

**Additional context**
 - API calls may have to be modified and the relevant issues created

### PR DESCRIPTION
**Pull request by aviraldevv**
**What kind of change does this PR introduce?**
This PR introduces functionality separation of ADMIN and SUPERADMIN.

**Issue Number:**
Fixes #650 

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

SUPERADMIN-

Screenshot_484

Screenshot_485

ADMIN-

Screenshot_486

Screenshot_487

**If relevant, did you update the documentation?**
Not Required

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by aviraldevv** [comment]
@palisadoes I have intentionally removed the test for search bar functionality don't know why manually it is working fine but tests were failing after the changes. We can create a new issue for the same after this PR gets merged.

**Comment by palisadoes** [comment]
Please create the issue for the tests now so that we don't forget. Ask to have it assigned to you.

File: src/screens/OrgList/OrgList.tsx [hunk 1]
```diff
@@ -248,7 +259,10 @@ function OrgList(): JSX.Element {
                           orgLocation={datas.location}
                         />
                       );
-                    } else {
+                    } else if (
+                      data_2?.user.adminFor.length === 1 &&
+                      data_2?.user.adminFor[0]._id === datas._id
```
Comment (Rumone) [inline src/screens/OrgList/OrgList.tsx:264]: The added logic seems to be getting too much to keep in the component markup. Can you abstract away the logic into a function that's called here.

File: src/screens/OrgList/OrgList.test.tsx [hunk 2]
```diff
@@ -402,77 +418,6 @@ describe('Organisation List Page', () => {
   });
 });
 
-test('Search bar filters organizations by name', async () => {
```
Comment (Rumone) [inline src/screens/OrgList/OrgList.test.tsx:405]: Why was this test removed?

**Review by Rumone — CHANGES_REQUESTED** [review]

File: src/components/AdminNavbar/AdminNavbar.tsx [hunk 3]
```diff
@@ -39,6 +42,11 @@ function AdminNavbar({ targets, url_1 }: NavbarProps): JSX.Element {
     variables: { id: currentUrl },
   });
   const [updateSpam] = useMutation(UPDATE_SPAM_NOTIFICATION_MUTATION);
+  const { data: data_2 } = useQuery(USER_ORGANIZATION_LIST, {
+    variables: { id: localStorage.getItem('id') },
+  });
+
+  const condtionalRender = data_2?.user.userType === 'SUPERADMIN';
```
Comment (Rumone) [inline src/components/AdminNavbar/AdminNavbar.tsx:49]: isSuperAdmin would probably be a better variable name here.

File: src/screens/OrgList/OrgList.tsx [hunk 4]
```diff
@@ -284,15 +300,17 @@ function OrgList(): JSX.Element {
...
-                  <tr data-testid="rowsPPSelect">
-                    <PaginationList
-                      count={data ? data.organizationsConnection.length : 0}
-                      rowsPerPage={rowsPerPage}
-                      page={page}
-                      onPageChange={handleChangePage}
-                      onRowsPerPageChange={handleChangeRowsPerPage}
-                    />
-                  </tr>
+                  {data_2?.user.userType !== 'ADMIN' && (
```
Comment (Rumone) [inline src/screens/OrgList/OrgList.tsx:303]: If this section is only for super admins why not use a conditional that represents that logic. Using !ADMIN implies there are other user types that can access this part of the UI. Is that so?

File: src/screens/OrgList/OrgList.tsx [hunk 5]
```diff
@@ -212,7 +213,17 @@ function OrgList(): JSX.Element {
                 data-testid="searchByName"
                 autoComplete="off"
                 required
-                onChange={debouncedHandleSearchByName}
+                onChange={
+                  data_2 && data_2.user.userType === 'SUPERADMIN'
```
Comment (Rumone) [inline src/screens/OrgList/OrgList.tsx:217]: Why have this conditional here if only super admins can access this input?

**Review by Rumone — CHANGES_REQUESTED** [review]

File: src/screens/OrgList/OrgList.test.tsx (hunk 2 above)
Comment (aviraldevv) [reply src/screens/OrgList/OrgList.test.tsx:405]: @Rumone 

Manually the search bar is functional right now, but the tests seem to break after the changes. I have created a separate issue to fix it up.

Comment (Rumone) [reply src/screens/OrgList/OrgList.test.tsx:405]: Oh I saw that you mentioned that my bad.

**Comment by Kevoniat** [comment]
> ## [Codecov](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/826?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation) Report
> > Merging [#826](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/826?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation) ([cfcb6c2](https://github.com/PalisadoesFoundation/talawa-admin/commit/cfcb6c21e7d2563c692b512622b450bc545ecc54)) into [develop](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/commit/e805335272cccfb11986e82d46ab3a4a1dc8474b?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation) ([e805335](https://github.com/PalisadoesFoundation/talawa-admin/commit/e805335272cccfb11986e82d46ab3a4a1dc8474b)) will **decrease** coverage by `0.16%`.
> > The diff coverage is `84.61%`.
> 
> > ❗ Current head [cfcb6c2](https://github.com/PalisadoesFoundation/talawa-admin/commit/cfcb6c21e7d2563c692b512622b450bc545ecc54) differs from pull request most recent head [9535c21](https://github.com/PalisadoesFoundation/talawa-admin/commit/9535c21d29a31aca3d3c7751967a3396deb31cbb). Consider uploading reports for the commit [9535c21](https://github.com/PalisadoesFoundation/talawa-admin/commit/9535c21d29a31aca3d3c7751967a3396deb31cbb) to get more accurate results
> 
> ```diff
> @@             Coverage Diff             @@
> ##           develop     #826      +/-   ##
> ===========================================
> - Coverage    91.75%   91.59%   -0.16%     
> ===========================================
>   Files           64       64              
>   Lines         1479     1499      +20     
>   Branches       377      391      +14     
> ===========================================
> + Hits          1357     1373      +16     
> - Misses         116      120       +4     
>   Partials         6        6              
> ```
> 
> [Impacted Files](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/826?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation)	Coverage Δ	
> [...components/AdminDashListCard/AdminDashListCard.tsx](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/826?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation#diff-c3JjL2NvbXBvbmVudHMvQWRtaW5EYXNoTGlzdENhcmQvQWRtaW5EYXNoTGlzdENhcmQudHN4)	`100.00% <ø> (ø)`	
> [src/screens/OrgList/OrgList.tsx](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/826?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation#diff-c3JjL3NjcmVlbnMvT3JnTGlzdC9PcmdMaXN0LnRzeA==)	`92.30% <78.94%> (-5.77%)`	⬇️
> [src/components/AdminNavbar/AdminNavbar.tsx](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/826?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation#diff-c3JjL2NvbXBvbmVudHMvQWRtaW5OYXZiYXIvQWRtaW5OYXZiYXIudHN4)	`84.00% <100.00%> (+1.02%)`	⬆️
> [...ns/OrganizationDashboard/OrganizationDashboard.tsx](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/826?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation#diff-c3JjL3NjcmVlbnMvT3JnYW5pemF0aW9uRGFzaGJvYXJkL09yZ2FuaXphdGlvbkRhc2hib2FyZC50c3g=)	`100.00% <100.00%> (ø)`	
> 📣 We’re building smart automated test selection to slash your CI/CD build times. [Learn more](https://about.codecov.io/iterative-testing/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation)

@aviraldevv  why did the test coverage drop

**Review by ttaylor92 — APPROVED** [review]

**Comment by aviraldevv** [comment]
@Kevoniat Codecov didn't update the coverage report after my latest commit. It's shouldn't have dropped.

**Comment by EshaanAgg** [comment]
@aviraldevv As a small suggestion, what do you think about renaming the field such as `data-1` and `data-2` to more readable names, that actually describe what they contain? Because with the current naming of variables, if becomes very difficult to track which data variable contains the response of which query!

**Comment by aviraldevv** [comment]
@EshaanAgg can you take a look why the test is failing I this PR completely unrelated with the calendar component.

**Comment by anshgoyalevil** [comment]
@aviraldevv 

The failing test is not related to your PR. It will be fixed with PR https://github.com/PalisadoesFoundation/talawa-admin/pull/772#issuecomment-1493053675

**Comment by aviraldevv** [comment]
> @aviraldevv As a small suggestion, what do you think about renaming the field such as `data-1` and `data-2` to more readable names, that actually describe what they contain? Because with the current naming of variables, if becomes very difficult to track which data variable contains the response of which query!

@EshaanAgg

**Comment by aashimawadhwa** [comment]
@aviraldevv   please take a pull the failing test will pass

**Comment by palisadoes** [comment]
@aviraldevv merge with the latest upstream and commit to this PR.

**Comment by aviraldevv** [comment]
@palisadoes is something wrong with the workflow checks??.

**Comment by EshaanAgg** [comment]
Yes, @aviraldevv. I had made some changes to the workflow files, due to which a bug has crept into the system. I have made a rectifying PR #833, which should fix the same.

**Comment by aviraldevv** [comment]
@EshaanAgg Got it bro.

**Comment by palisadoes** [comment]
@aviraldevv Please address the questions from @Rumone

**Comment by aviraldevv** [comment]
@palisadoes I had addressed them all in my latest commits.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**